### DISCUSSION TITLE
Worker scale horizontally

### DISCUSSION THREAD

**Discussion by ndTEC** [postalserver/postal#2934] [Help with using Postal]
In case we scale de worker using docker-compose.override.yml how could we manage the health port for each worker?

**Comment by willpower232** [comment]
Do you mean a third party service making sure your Postal server is still working? I think your server would still answer port 25 so it would work.

You could have something counting processes on the server as well I guess.

**Reply by ndTEC** [reply]
Each container is configured with network: host, therefore port 9090 can be used only by one worker container. When running multiple worker containers the health server is started only on the first one.

health server port (127.0.0.1:9090) is already in use, not starting health server

**Reply by willpower232** [reply]
oh you're trying to monitor that all the workers are running? not just that the server has at least one worker?

**Reply by ndTEC** [reply]
I want to scrape prometheus metrics from all workers.

**Comment by ndTEC** [comment]
I would recommend for the workers to test the health server port and if not available to increment.
